app/src/main/java/ch/epfl/skysync/screens/ConfirmFlightUI.kt

```diff
@@ -0,0 +1,415 @@
+        OutlinedButton(onClick = { expanded = true }) { Text(text = selected) }
+
+        DropdownMenu(expanded = expanded, onDismissRequest = { expanded = false }) {
+          options.forEachIndexed { index, option ->
+            DropdownMenuItem(
+                onClick = {
+                  selectedIndex = index
+                  selectedOption = option
+                  expanded = false
+                  onOptionSelected(option)
+                },
+                modifier = Modifier.padding(vertical = 1.dp).testTag(option)) {
+                  Text(text = option)
+                }
+          }
+        }
+
+        selectedOption?.let { selected = it }
+      }
+}
+/**
+ * Composable function to create an element with a text input field, a button to add the input as a
+ * remark, and a list of remarks.
+ *
+ * @param title The title of the element. Defaults to "Remark".
+ * @param remarksList A lambda function that provides the list of remarks.
+ * @param onAddRemark A lambda function to handle adding a remark. Accepts a string parameter
+ *   representing the remark to add.
+ */
+@Composable
+fun AddElementComposable(
+    title: String = "Remark",
+    remarksList: () -> List<String>,
+    onAddRemark: (String) -> Unit
+) {
+  var remarkText by remember { mutableStateOf("") }
+
+  Column(
+      modifier = Modifier.padding(0.dp).fillMaxWidth(),
+      horizontalAlignment = Alignment.CenterHorizontally) {
+        TextField(
+            value = remarkText,
+            onValueChange = { remarkText = it },
+            label = { Text("Enter " + title) },
+            keyboardOptions = KeyboardOptions.Default.copy(imeAction = ImeAction.Done),
+            keyboardActions =
+                KeyboardActions(
+                    onDone = {
+                      onAddRemark(remarkText)
+                      remarkText = ""
+                    }),
+            modifier = Modifier.fillMaxWidth().padding(bottom = 0.dp))
+
+        Button(
+            onClick = {
+              onAddRemark(remarkText)
+              remarkText = ""
+            },
+            colors = ButtonDefaults.buttonColors(backgroundColor = lightOrange)) {
+              Text("Add " + title)
+            }
+
+        Column {
+          remarksList().forEach { remark ->
+            if (remark != "") {
+              Text(text = remark)
+            }
+          }
+        }
+      }
+}
+//    @Composable
+//    @Preview()
+//    fun confirmationScreenPreview(){
+//        val dummy = PlannedFlight(
+//            "1234",
+//            3,
+//            FlightType.DISCOVERY,
+//            Team(listOf(Role(RoleType.CREW))),
+//            Balloon("Balloon Name", BalloonQualification.LARGE,"Ballon Name"),
+//            Basket("Basket Name",true,"1234"),
+//            LocalDate.now().plusDays(3),
+//            TimeSlot.PM,
+//            listOf(Vehicle("Peugeot 308","1234")))
+//        confirmationScreen(dummy)
+//    }
```

Comment:
you could remove this instead of commenting it

Reply:
I did it like @Plouc314 did it when he refactored my code